### DISCUSSION TITLE
Provide an example using the Hooks version with SSR and routing in remix

### DISCUSSION THREAD

**Discussion by wintersieck** [algolia/instantsearch#5362] [General] [answered]
We're struggling to get the new Hooks version of the library working with SSR and the `history` router in [remix](https://remix.run/). We can generally get it wired up following the NextJS example, but using the back/forward button causes it to either break entirely or not keep state correctly. The main issue is that remix seems to handle routing differently and re-renders when the URL changes, which causes `InstantSearchSSRProvider` to try to re-use the original `serverState`.

We would be very appreciative if you could create an example on [your SSR examples page](https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react-hooks/) similar to [this](https://codesandbox.io/s/github/algolia/react-instantsearch/tree/v6.29.0/examples/hooks-next?file=/pages/index.tsx:969-982), but for remix.

Thank you!

**Comment by francoischalifour** [comment, marked as answer]
Hi @wintersieck, we've fixed a few lifecycle issues with the library in [v6.30.2](https://github.com/algolia/react-instantsearch/releases/tag/v6.30.2). Could you please upgrade and let us know how it goes?

We'll keep you posted when we create a dedicated Remix example.

**Reply by francoischalifour** [reply]
We created a [Remix example](https://github.com/algolia/react-instantsearch/blob/d50a5807eb8a9d82412fbdf39b52e939a68f19ee/examples/hooks-remix/app/routes/index.tsx) in the repo. Let us know if it helps.

**Reply by wintersieck** [reply]
Thank you @francoischalifour! We'll review it shortly.

**Reply by wintersieck** [reply]
Thanks again for the example @francoischalifour! We tested in our app and logged #3580.
